### ISSUE TITLE
Recommended autoformatter introduces lint errors

### ISSUE THREAD

**Issue by ggoretkin-bdai** [jumanjihouse/pre-commit-hook-yamlfmt#48]
For YAML files like this:

```yaml
location: external
sources:
  - repo: foo
    name: bar
    rev: main
    type: git
    params:
    sparse_paths:
      -
    links:
      -
```

The autoformatter adds trailing spaces:

```diff
diff --git a/test.yaml b/test.yaml
index 272e7eb..f1b8fbe 100644
--- a/test.yaml
+++ b/test.yaml
@@ -1,12 +1,13 @@
+---
 location: external
 sources:
-  - repo: foo
-    name: bar
-    rev: main
-    type: git
-    params:
-    sparse_paths:
-      -
-    links:
-      -
+    - repo: foo
+      name: bar
+      rev: main
+      type: git
+      params:
+      sparse_paths:
+          - 
+      links:
+          - 
```

![image](https://user-images.githubusercontent.com/124000349/235828604-9bb8c11a-7e76-4a59-a0cb-e793516e959d.png)

**Comment by jumanjiman** [comment]
**temp** workaround: `additional_dependencies: [ruamel.yaml==0.17.21]`

**Comment by jumanjiman** [comment]
possibly related to https://github.com/jumanjihouse/pre-commit-hook-yamlfmt/issues/47


from https://sourceforge.net/p/ruamel-yaml/code/ci/0.17.22/tree/CHANGES

> plain scalars: put single words longer than width on a line of their own, instead
    of after the previous line (issue 427, reported by `Antoine Cotten 
    <https://sourceforge.net/u/antoineco/profile/>`_). Caveat: this currently results in a 
    space ending the previous line.

**Comment by jumanjiman** [comment]
published new tag `0.2.3` to constrain ruamel.yaml version: 
https://github.com/jumanjihouse/pre-commit-hook-yamlfmt/releases/tag/0.2.3
